dask_expr/io/parquet.py

```diff
@@ -379,25 +397,70 @@ def to_parquet(
     return out
 
 
-class ReadParquet(PartitionsFiltered, BlockwiseIO):
-    """Read a parquet dataset"""
+# class ReadParquetPyarrowFSNoMeta(ReadParquetPyarrowFS):
+
+#     def _lower(self):
+#         return ReadParquetPyarrowFS(
+
+
+#         )
+def _determine_type_mapper(
+    *, user_types_mapper=None, dtype_backend=None, convert_string=True
+):
+    type_mappers = []
+
+    def pyarrow_type_mapper(pyarrow_dtype):
+        # Special case pyarrow strings to use more feature complete dtype
+        # See https://github.com/pandas-dev/pandas/issues/50074
+        if pyarrow_dtype == pa.string():
+            return pd.StringDtype("pyarrow")
+        else:
+            return pd.ArrowDtype(pyarrow_dtype)
+
+    # always use the user-defined mapper first, if available
+    if user_types_mapper is not None:
+        type_mappers.append(user_types_mapper)
 
+    # next in priority is converting strings
+    if convert_string:
+        type_mappers.append({pa.string(): pd.StringDtype("pyarrow")}.get)
+        type_mappers.append({pa.date32(): pd.ArrowDtype(pa.date32())}.get)
+        type_mappers.append({pa.date64(): pd.ArrowDtype(pa.date64())}.get)
+
+        def _convert_decimal_type(type):
+            if pa.types.is_decimal(type):
+                return pd.ArrowDtype(type)
+            return None
+
+        type_mappers.append(_convert_decimal_type)
+
+    # and then nullable types
+    if dtype_backend == "numpy_nullable":
+        type_mappers.append(PYARROW_NULLABLE_DTYPE_MAPPING.get)
+    elif dtype_backend == "pyarrow":
+        type_mappers.append(pyarrow_type_mapper)
+
+    def default_types_mapper(pyarrow_dtype):
+        """Try all type mappers in order, starting from the user type mapper."""
+        for type_converter in type_mappers:
+            converted_type = type_converter(pyarrow_dtype)
+            if converted_type is not None:
+                return converted_type
+
+    if len(type_mappers) > 0:
+        return default_types_mapper
+
+
+class ReadParquetPyarrowFS(PartitionsFiltered, BlockwiseIO):
```

Comment:
I just noticed that this is not inheriting from `ReadParquet`. I intended this to inherit since there are many places in the code path where we're checking for `ReadParquet`. So.... this didn't use any of the optimizations defined in  `ReadParquet.simplify_up`. I guess those are not very relevant for TPCH? 🤔